aea/helpers/exec_timeout.py

```diff
@@ -273,4 +271,4 @@ def _remove_timeout_watch(self) -> None:
         """
         if self._future_guard_task and not self._future_guard_task.done():
             self._future_guard_task.cancel()
-            self._future_guard_task = None
+        self._future_guard_task = None
```

Comment:
this did not appears to resolve the issue, as various CI runs failed just the same